### ISSUE TITLE
NF repeats warning "No more task to compute -- Execution may be stalled" after all tasks killed by scheduler

### ISSUE THREAD

**Issue by Shellfishgene** [nextflow-io/nextflow#407]
The final tasks of the workflow were killed by the scheduler due to time overrun, but NF did not quit but repeated the above warning every 5 minutes until manually stopped with ctrl-c.

[.nextflow.log](https://github.com/nextflow-io/nextflow/files/1169360/ssxeiUGq.txt)

**Comment by pditommaso** [comment]
Though does not explain the issue, this error in the log file may be relevant on your side 

```
ERROR nextflow.processor.TaskProcessor - Error executing process > 'report (1)'

Caused by:
  Process `report (1)` terminated with an error exit status (1)

Command executed:

  ~/programs/sga/bin/sga-preqc-report.py F09791.preqc ~/programs/sga/examples/preqc/
  mv preqc_report.pdf F09791.report.pdf

Command exit status:
  1

Command output:
  output_pfx:  preqc_report
  preqc_files:
  	F09791.preqc
  	/sfs/fs6/home-geomar/smomw240/programs/sga/examples/preqc/

Command error:
  Traceback (most recent call last):
    File "/sfs/fs6/home-geomar/smomw240/programs/sga/bin/sga-preqc-report.py", line 868, in <module>
      sys.exit(main(argv=None))
    File "/sfs/fs6/home-geomar/smomw240/programs/sga/bin/sga-preqc-report.py", line 88, in main
      data = load_preqc_datafiles(preqc_files)
    File "/sfs/fs6/home-geomar/smomw240/programs/sga/bin/sga-preqc-report.py", line 696, in load_preqc_datafiles
      deserial = json.load(open(f, 'r'))
  IOError: [Errno 21] Is a directory: '/sfs/fs6/home-geomar/smomw240/programs/sga/examples/preqc'

Work dir:
  /sfs/fs3/work-geomar7/smomw240/luisa_seq/work/08/ceb6bdfcbfd7795bdf94f3c6a9a874
```

**Comment by pditommaso** [comment]
I've uploaded a new snapshot, tho I don't think it will solve the problem, it should report more detailed information. Could you give it a try ? You will need to declare the following variable 

```
export NXF_VER=0.25.3-SNAPSHOT
```
then use NF as usual

**Comment by pditommaso** [comment]
Is this issue still open? if yes, have you tried with the latest snapshot as suggsted  ?

**Comment by Shellfishgene** [comment]
Still open, have to make a test case as the one I had this with runs too long.

**Comment by Shellfishgene** [comment]
I can't really replicate it now, I copied the same workflow but replaced all commands with `touch $file` or similar. The one that was killed previously I added `wait 240` and set the scheduler timeout to 1 min. Now nf recognizes what happens and gives the `Likely it has been terminated by the external system`.
That still takes very long after the scheduler terminates the tasks though, ca. 10 minutes.
